### PR TITLE
PIM-5681: Add version purger documentation

### PR DESCRIPTION
**Pull request by SamirBoulil**

### CODE REVIEW

**Comment by loicguillois** [comment]
👍

File: cookbook/purge/index.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+How to purge history
+====================
+
+This chapter describes the existing commands that exists to purge unused information in the PIM.
```
Comment (MarieMinasyan) [inline cookbook/purge/index.rst]: "the existing commands" is enough, please remove "that exists"

File: cookbook/purge/purge_version_entities.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
+How to customize the version purger to your needs
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: "How to customize the version purger" or "How to adapt the version purger to your needs"

File: cookbook/purge/purge_version_entities.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+=================================================
+
+After an intensive use of the PIM, it is possible that your database's volume has increased in a way that the PIM holds too many unused versions.
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: database volume

File: cookbook/purge/purge_version_entities.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+It is possible to purge unused versions in the PIM for a specific entity or for any entities using the version purger (see the command ``pim:versioning:purge``).
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: or for all entities

File: cookbook/purge/purge_version_entities.rst [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+In this cookbook, we will go through each step to customize this command's behavior. It is built to be easily extended to keep versions of your interest.
+
+We are going to customize the version purger to keep versions which number is pair.
+
+Version Purger's structure
+--------------------------
+
+The version purger is able to retrieve the versions from the database and to run them against a list of advisors to know if they need to be purged.
+
+The advisor's responsibility is to indicate wether a version should be purged or not.
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: whether

File: cookbook/purge/purge_version_entities.rst [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+ - If all advisors registered within the version purger agree on purging, the version will be removed
+ - If one advisor indicates the version should be kept it is not removed
+
+We can customize the purge process by registering a new advisor within the version purger.
+
+
+Custom advisor : SkipPairVersionAdvisor
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: advisor:

File: cookbook/purge/purge_version_entities.rst [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            class: '%acme_app.version_purger.skip_pair_version.class%'
+            tags:
+                - { name: pim_versioning.purger.advisor, priority: 100 }
+
+
+.. note::
+
+    Do not forget to load the ``purger_advisors.yml`` configuration file in your bundle as well as clear the symfony cache.
+
+Here we go ! We can now run the purge command to observe that our advisor is correctly registered in the version purger.
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: go!

File: cookbook/purge/purge_version_entities.rst [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+After an intensive use of the PIM, it is possible that your database's volume has increased in a way that the PIM holds too many unused versions.
+
+It is possible to purge unused versions in the PIM for a specific entity or for any entities using the version purger (see the command ``pim:versioning:purge``).
+
+By default, the command will purge any version older than 90 days and that is not the first one nor the last one for the entity.
+
+In this cookbook, we will go through each step to customize this command's behavior. It is built to be easily extended to keep versions of your interest.
+
+We are going to customize the version purger to keep versions which number is pair.
```
Comment (MarieMinasyan) [inline cookbook/purge/purge_version_entities.rst]: versions whose number is pair

**Comment by SamirBoulil** [comment]
Awesome thanks @MarieMinasyan !

File: cookbook/purge/purge_version_entities.rst [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+    Do not forget to load the ``purger_advisors.yml`` configuration file in your bundle as well as clear the symfony cache.
+
+Here we go! We can now run the purge command to observe that our advisor is correctly registered in the version purger.
+
+You can add a few test versions in the PIM by modifying and saving a product or a family for instance.
+
+.. code-block:: bash
+
+    app/console pim:versioning:purge --more-than-days 0  --env=prod
```
Comment (pierallard) [inline cookbook/purge/purge_version_entities.rst]: double space

File: cookbook/purge/purge_version_entities.rst [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+The advisor's responsibility is to indicate whether a version should be purged or not.
+
+ - If all advisors registered within the version purger agree on purging, the version will be removed
+ - If one advisor indicates the version should be kept it is not removed
+
+We can customize the purge process by registering a new advisor within the version purger.
+
+
+Advisor : SkipPairVersionAdvisor
```
Comment (pierallard) [inline cookbook/purge/purge_version_entities.rst]: remove space after Advisor

File: cookbook/purge/purge_version_entities.rst [hunk 11]
```diff
@@ -0,0 +1,107 @@
+How to adapt the version purger to your needs
+=============================================
+
+After an intensive use of the PIM, it is possible that your database volume has increased in a way that the PIM holds too many unused versions.
```
Comment (pierallard) [inline cookbook/purge/purge_version_entities.rst]: your database volume may have increased?

Comment (SamirBoulil) [inline cookbook/purge/purge_version_entities.rst]: 'possible' here already gives the 'conditionnal' here no need for a may :)

File: cookbook/purge/purge_version_entities.rst [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        {
+            // Our advisor supports any kind of versionable entity
+            return true;
+        }
+
+        /**
+         * Indicates if the version needs to be purged
+         *
+         * @param VersionInterface $version
+         * @param array $options
```
Comment (pierallard) [inline cookbook/purge/purge_version_entities.rst]: align :)

File: cookbook/purge/purge_version_entities.rst (hunk 11 above)
Comment (pierallard) [inline cookbook/purge/purge_version_entities.rst]: Yes but the next sentence have "it is possible" too in the other sence... It's confusing.

**Comment by pierallard** [comment]
Little comment then GTM maggle

File: cookbook/purge/purge_version_entities.rst (hunk 12 above)
Comment (SamirBoulil) [inline cookbook/purge/purge_version_entities.rst]: Even in sample code ? Ok :)

**Pull request merged**